socorro/external/postgresql/setupdb_app.py

```diff
@@ -235,120 +235,62 @@ def create_connection_url(self, database_name, username, password):
             url += '/%s' % database_name
         return url
 
-    def main(self):
+    def handle_dropdb(self, superuser_pg_url, database_name):
+        """Handles dropping the database
 
-        database_name = self.config.database_name
-        if not database_name:
-            self.config.logger.error(
-                '"database_name" cannot be an empty string'
-            )
-            return 1
+        If the database is not a test database, this will prompt the user to make sure the user is
+        ok with dropping the database.
 
-        # superuser credentials for overall database
-        superuser_pg_url = self.create_connection_url(
-            'postgres',
-            self.config.database_superusername,
-            self.config.database_superuserpassword
-        )
+        :arg string superuser_pg_url: the postgres url for the superuser user
+        :arg string database_name: the name of the database to drop
 
-        # superuser credentials for working database
-        superuser_normaldb_pg_url = self.create_connection_url(
-            database_name,
-            self.config.database_superusername,
-            self.config.database_superuserpassword
-        )
+        :returns: True if everything is good, False if the user said "don't drop this"
 
-        # normal user credentials
-        normal_user_pg_url = self.create_connection_url(
-            database_name,
-            self.config.database_username,
-            self.config.database_password
-        )
+        """
+        self.config.logger.info('drop database section with %s', superuser_pg_url)
+        with PostgreSQLAlchemyManager(
+                superuser_pg_url,
+                self.config.logger,
+                autocommit=False
+        ) as db:
+            if 'test' not in database_name and not self.config.force:
+                confirm = raw_input(
+                    'drop database %s [y/N]: ' % database_name
+                )
+                if not confirm == "y":
+                    self.config.logger.warning('NOT dropping table')
+                    return False
+            db.drop_database(database_name)
+            db.commit()
+        return True
```

Comment:
This code was moved from `main`. The only thing that changed here is that it returns `True` and `False` rather than integers.